modules/reactor/src/meshgenerators/TriPinHexAssemblyGenerator.C

```diff
@@ -68,6 +68,15 @@ TriPinHexAssemblyGenerator::validParams()
   params.addParam<std::vector<dof_id_type>>(
       "pin_id_values",
       "Values of extra integer ID to be assigned to each of the three pin domains.");
+  params.addParam<std::string>("sector_id_name",
+                               "Name of integer (reporting) ID for sector regions to use the "
+                               "reporting ID for azimuthal sector regions of ring geometry block.");
+  params.addParam<std::string>("ring_id_name",
+                               "Name of integer (reporting) ID for ring regions to use the "
+                               "reporting ID for annular regions of ring geometry block.");
+  MooseEnum ring_id_option("block_wise ring_wise", "block_wise");
+  params.addParam<MooseEnum>(
+      "ring_id_assign_type", ring_id_option, "Type of ring ID assignment: block_wise or ring_wise");
```

Comment:
these parameters could be added by a routine defined in a base class to be shared with the other MG

also add a parameter group for ring/sector IDs

Reply:
In the base class, I added a routine to append input parameters for ring / sector IDs as :
```c++
void
PolygonMeshGeneratorBase::addRingAndSectorIDParams(InputParameters & params)
{
  params.addParam<std::string>("sector_id_name",
                               "Name of integer (reporting) ID for sector regions to use the "
                               "reporting ID for azimuthal sector regions of ring geometry block.");
  params.addParam<std::string>("ring_id_name",
                               "Name of integer (reporting) ID for ring regions to use the "
                               "reporting ID for annular regions of ring geometry block.");
  MooseEnum ring_id_option("block_wise ring_wise", "block_wise");
  params.addParam<MooseEnum>(
      "ring_id_assign_type", ring_id_option, "Type of ring ID assignment: block_wise or ring_wise");
  params.addParamNamesToGroup("sector_id_name ring_id_name ring_id_assign_type", "Ring/Sector IDs");
}
```